consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/jsonrpc/methods/QbftGetValidatorsByBlockNumber.java

```diff
@@ -53,6 +53,21 @@ protected BlockParameter blockParameter(final JsonRpcRequestContext request) {
     return request.getRequiredParameter(0, BlockParameter.class);
   }
 
+  @Override
+  protected Object pendingResult(final JsonRpcRequestContext request) {
+    final long blockNumber = blockchainQueriesSupplier.get().headBlockNumber();
```

Comment:
nit: could do this since the block number is only being used to get the head block header
```suggestion
    final BlockHeader blockHeader = getBlockchainQueries().headBlockHeader();
```

Reply:
There's also the log entry, but that can be replaced with `blockHeader.getNumber()` so seems fine to me.